pkg/cmd/opts/install.go

```diff
@@ -777,10 +777,38 @@ func (o *CommonOptions) InstallMinishift() error {
 
 // InstallGcloud installs gcloud cli
 func (o *CommonOptions) InstallGcloud() error {
-	if runtime.GOOS != "darwin" || o.NoBrew {
-		return errors.New("please install missing gcloud sdk - see https://cloud.google.com/sdk/downloads#interactive")
+	versionResolver, err := o.GetVersionResolver()
+	if err != nil {
+		return err
+	}
+	stableVersion, err := versionResolver.StableVersionNumber(versionstream.KindPackage, "gcloud")
+	if err != nil {
+		return err
 	}
-	return o.RunCommand("brew", "cask", "install", "google-cloud-sdk")
+	clientURL := fmt.Sprintf("https://storage.googleapis.com/cloud-sdk-release/google-cloud-sdk-%s-%s-x86.tar.gz", stableVersion, runtime.GOOS)
+	if strings.Contains(runtime.GOARCH, "64") {
+		clientURL = fmt.Sprintf("https://storage.googleapis.com/cloud-sdk-release/google-cloud-sdk-%s-%s-x86_64.tar.gz", stableVersion, runtime.GOOS)
+	}
+	binDir, err := util.JXBinLocation()
+	binary := "gcloud"
+	if err != nil {
+		return err
+	}
+	fileName, flag, err := packages.ShouldInstallBinary(binary)
```

Comment:
This one is tricky. @nehagup was also running into it. We don't want to use _ShouldInstallBinary_  here. For some reason it checks generally in the global PATH whether a given binary is installed. This means if you have any the binary installed anywhere, it would already return _false_. Our aim it to install into _JX_HOME/bin_ if it does not exist there, independent of other versions of the binary on the PATH.

```go
	pgmPath, err := exec.LookPath(fileName)
	if err == nil {
		log.Logger().Debugf("%s is already available on your PATH at %s", util.ColorInfo(fileName), util.ColorInfo(pgmPath))
		return
	}
```

I think we could remove the above code from  `ShouldInstallBinary` to rectify this in general, but that might affect some other callers of this method. 

The alternative is what @nehagup has done for installing `kustomize` - https://github.com/jenkins-x/jx/blob/master/pkg/cmd/opts/install.go#L237
One just checks whether the binary exists at the expected path.

Reply:
Yeah I guess checking if the file exists in the `JX_HOME/bin` makes sense, rather than checking the whole path. The function `ShouldInstallBinary` can be changed later if required.